fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageUtils.java

```diff
@@ -170,21 +175,58 @@ public static RdfStream getRdfStream(final String identifier,
         }
     }
 
-    private static  String resolveVersionId(final OCFLObjectSession objSession, final Instant version) {
-        //TODO Implement resolution of a version id (OCFL-speak) from an instant (memento-speak)
-       return null;
+    private static String resolveVersionId(final OCFLObjectSession objSession, final Instant version)
+            throws PersistentStorageException {
+        if (version != null) {
+            return objSession.listVersions()
+                    .stream()
+                    .filter(vd -> {
+                        //filter by comparing Epoch seconds since
+                        //Memento has second granularity while OCFL versions are
+                        //millisecond granularity
+                        return vd.getCreated()
+                                .toInstant()
+                                .toEpochMilli() / 1000 == version.toEpochMilli() / 1000;
```

Comment:
So this appears to match for a version created exactly at the Instant. Current functionality is to give you the closest, but older version if it exists. Does that work happen at a higher level or should this filtering also do that?

Reply:
I was thinking that the "nearest" match would be resolved at a higher level using the the listVersions(
String fedoraId) method.